security/medusa/l3/registry.c

```diff
@@ -197,182 +158,186 @@ int med_unregister_kclass(struct medusa_kclass_s * ptr)
 	MED_UNLOCK_R(usecount_lock);
 	MED_UNLOCK_R(registry_lock);
 	if (authserver && authserver->del_kclass)
-		authserver->del_kclass(ptr);
+		authserver->del_kclass(med_kclass);
 	/* FIXME: this isn't safe. add use-count to authserver too... */
 	return 0;
 }
 
 /**
  * med_register_kclass - register a kclass of k-objects and notify the authserver
- * @ptr: pointer to the kclass to register
+ * @med_kclass: pointer to the kclass to register
  *
  * The authserver call must be in lock or a semaphore - we promised
  * that in authserver.h. :)
  */
-int med_register_kclass(struct medusa_kclass_s * ptr)
+int med_register_kclass(struct medusa_kclass_s *med_kclass)
 {
-	struct medusa_kclass_s * p;
+	struct medusa_kclass_s *p;
 
-	ptr->name[MEDUSA_KCLASSNAME_MAX-1] = '\0';
-	med_pr_info("Registering kclass %s\n", ptr->name);
+	med_kclass->name[MEDUSA_KCLASSNAME_MAX-1] = '\0';
+	med_pr_info("Registering kclass %s\n", med_kclass->name);
 	MED_LOCK_W(registry_lock);
-	for (p=kclasses; p; p=p->next)
-		if (ptr==p || !mystrcmp(p->name, ptr->name)) {
-			med_pr_err("Error: such kclass already exists.\n");
+	for (p = kclasses; p; p = p->next)
+		if (med_kclass == p) {
```

Comment:
There cannot be two kclasses with the same name! so, we should ensure that this cannot happen.
see old test: `if (ptr==p || !mystrcmp(p->name, ptr->name))`
the second part should be used, too!